modules/cas_cache/layer_cache_management.c

```diff
@@ -2385,11 +2397,31 @@ int cache_mngt_init_instance(struct ocf_mngt_cache_config *cfg,
 	switch (cmd->init_cache) {
 	case CACHE_INIT_LOAD:
 	case CACHE_INIT_STANDBY_LOAD:
-		result = _cache_mngt_check_metadata(cfg, cmd->cache_path_name);
+		result = _cache_mngt_probe_metadata(cmd->cache_path_name,
+				cache_name_meta, &cache_mode_meta,
+				&cache_line_size_meta);
 		if (result) {
 			module_put(THIS_MODULE);
 			return result;
 		}
+
+		result = ocf_mngt_cache_get_by_name(cas_ctx, cache_name_meta,
+				OCF_CACHE_NAME_SIZE, &tmp_cache);
+		/* Need to return name from metadata now for caller to properly
+		 * communicate the error to user */
+		BUG_ON(cache_id_from_name(&cmd->cache_id, cache_name_meta));
+
+		if (result != -OCF_ERR_CACHE_NOT_EXIST) {
```

Comment:
I think any other result != 0 should also be treated as error

Reply:
This check verifies if there's already any cache started and running with the same name that we're trying to load to avoid name collision. The error condition is here everything that isn't `OCF_ERR_CACHE_NOT_EXIST`.